includes/handlers/class-webmention-handler-mf2.php

```diff
@@ -529,4 +533,200 @@ public function compare_urls( $needle, $haystack, $schemeless = true ) {
+		 */
+		$class_mapper['in-reply-to'] = 'reply';
+		$class_mapper['reply']       = 'reply';
+		$class_mapper['reply-of']    = 'reply';
+
+		/*
+		 * repost
+		 * @link http://indieweb.org/repost
+		 */
+		$class_mapper['repost']    = 'repost';
+		$class_mapper['repost-of'] = 'repost';
+
+		/*
+		 * likes
+		 * @link http://indieweb.org/likes
+		 */
+		$class_mapper['like']    = 'like';
+		$class_mapper['like-of'] = 'like';
+
+		/*
+		 * favorite
+		 * @link http://indieweb.org/favorite
+		 */
+		$class_mapper['favorite']    = 'favorite';
+		$class_mapper['favorite-of'] = 'favorite';
+
+		/*
+		 * bookmark
+		 * @link http://indieweb.org/bookmark
+		 */
+		$class_mapper['bookmark']    = 'bookmark';
+		$class_mapper['bookmark-of'] = 'bookmark';
+
+		/*
+		 * rsvp
+		 * @link http://indieweb.org/rsvp
+		 */
+		$class_mapper['rsvp'] = 'rsvp';
+		/*
+		 * invite
+		 * @link https://indieweb.org/invitation
+		 */
+		$class_mapper['invitee'] = 'invite';
+
+		/*
+		 * tag
+		 * @link http://indieweb.org/tag
+		 */
+		$class_mapper['tag-of']   = 'tag';
+		$class_mapper['category'] = 'tag';
+
+		/*
+		 * read
+		 * @link http://indieweb.org/read
+		 */
+		$class_mapper['read-of'] = 'read';
+		$class_mapper['read']    = 'read';
+
+		/*
+		 * listen
+		 * @link http://indieweb.org/listen
+		 */
+		$class_mapper['listen-of'] = 'listen';
+		$class_mapper['listen']    = 'listen';
+
+		/*
+		 * watch
+		 * @link http://indieweb.org/watch
+		 */
+		$class_mapper['watch-of'] = 'watch';
+		$class_mapper['watch']    = 'watch';
+
+		/*
+		 * follow
+		 * @link http://indieweb.org/follow
+		 */
+		$class_mapper['follow-of'] = 'follow';
+
+		return apply_filters( 'webmention_microformats_class_mapper', $class_mapper );
+	}
+
+	/**
+	 * All supported url types
+	 *
+	 * @return array
+	 */
+	protected function get_rel_mapper() {
+		$rel_mapper = array();
+
+		/*
+		 * replies
+		 * @link http://indieweb.org/in-reply-to
+		 */
+		$rel_mapper['in-reply-to'] = 'reply';
+		$rel_mapper['reply-of']    = 'reply';
+
+		/*
+		 * bookmarks
+		 * @link http://microformats.org/wiki/rel-design-pattern#rel.3D.22bookmark.22
+		 */
+		$rel_mapper['bookmark'] = 'bookmark';
+
+		/*
+		 * tags
+		 * @link http://microformats.org/wiki/rel-tag
+		 */
+		$rel_mapper['tag'] = 'tag';
+
+		return apply_filters( 'webmention_microformats_rel_mapper', $rel_mapper );
+	}
+
+	/**
+	 * check entry classes or document rels for post-type
+	 *
+	 * @param string $target the target url
+	 * @param array $entry the represantative entry
+	 * @param array $mf_array the document
+	 *
+	 * @return string the post-type
+	 */
+	protected function get_response_type( $entry, $mf_array, $target ) {
+		$classes = $this->get_class_mapper();
+
+		// check properties for target-url
+		foreach ( $entry['properties'] as $key => $values ) {
+			// check u-* params
+			if ( in_array( $key, array_keys( $classes ), true ) ) {
+				// check RSVP if available
+				if ( 'rsvp' === $key ) {
```

Comment:
The rsvp:value...do we want to retain the : typing for rsvp? It's the only scenario where the the value of the property dictates the type, I believe.

Reply:
good point! what about adding the sub-type as comment body?

Reply:
I'm still not sure what you meant about this...As in, it goes in content? That would make sense, I suppose. We'd need to do that outside of this function though.